src/contour/ContourGuiApp.cpp

```diff
@@ -149,7 +149,17 @@ crispy::cli::command ContourGuiApp::parameterDefinition() const
 
 std::chrono::seconds ContourGuiApp::earlyExitThreshold() const
 {
-    return std::chrono::seconds(parameters().get<unsigned>("contour.terminal.early-exit-threshold"));
+    auto const configThreshold = config().earlyExitThreshold.value();
+    auto const parameterThreshold = parameters().get<unsigned>("contour.terminal.early-exit-threshold");
+
+    // default threshold is config::documentation::DefaultEarlyExitThreshold seconds
+    if (parameterThreshold != config::documentation::DefaultEarlyExitThreshold)
+        return std::chrono::seconds(parameterThreshold);
```

Comment:
what if the user actually explicitly passed an early-exit-threshold that matches the default? Then this branch would not be taken.

Reply:
You are right, updated implementation